EIPS/eip-7498.md

```diff
@@ -0,0 +1,299 @@
+
+### Redemption extraData
+
+When calling the `redeem` function, the extraData layout MUST follow:
+
+| bytes    | value             | description / notes                                              |
+| -------- | ----------------- | ---------------------------------------------------------------- |
+| 0-32     | campaignId        |                                                                  |
+| 32-64    | redemptionHash    | hash of offchain order ids                                       |
+| 64-\*    | TraitRedemption[] | see TraitRedemption struct. empty array for no trait redemptions |
+| \*-(+32) | salt              | if signer != address(0)                                          |
+| \*-(+\*) | signature         | if signer != address(0). can be for EIP-712 or ERC-1271          |
+
+Upon redemption, the contract MUST check that the campaign is still active (using the same boundary check as Seaport, `startTime <= block.timestamp < endTime`). If it is, it MUST revert with `NotActive()`.
+
+### Redeem
+
+The `redeem` function MUST execute the transfers in the `consideration`. It MUST also call `mintRedemption` on the token specified in the `offer`. If any of the supplied tokenIds in `redeem` fail validation, the function MAY execute just the redemptions that were valid and ignore the failed redemptions. The `Redemption` event MUST be emitted emitted when any valid redemptions occur.
+
+### Trait redemptions
+
+The token MUST respect the TraitRedemption substandards as follows:
+
+| substandard ID | description                     | substandard value                                                  |
+| -------------- | ------------------------------- | ------------------------------------------------------------------ |
+| 1              | set value to `traitValue`       | prior required value. if blank, cannot be the `traitValue` already |
+| 2              | increment trait by `traitValue` | max value                                                          |
+| 3              | decrement trait by `traitValue` | min value                                                          |
+
+### Max campaign redemptions
+
+The token MUST check that the `maxCampaignRedemptions` is not exceeded. If the redemption does exceed `maxCampaignRedemptions`, it MUST revert with `MaxCampaignRedemptionsReached(uint256 total, uint256 max)`
+
+### Metadata URI
+
+The metadata URI MUST follow the following JSON schema:
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string"
+    },
+    "description": {
+      "type": "string",
+      "description": "A one-line summary of the redeemable. Markdown is not supported."
+    },
+    "details": {
+      "type": "string",
+      "description": "A multi-line or multi-paragraph description of the details of the redeemable. Markdown is supported."
+    },
+    "imageUrls": {
```

Comment:
Why not include an "image" key so these NFTs don't break other places where NFTs are displayed? The image key can then be used as a thumbnail. The metadata format URI should be a superset of normal NFT metadata.

Reply:
ah yes it should be an array thanks, will fix. the first image should be used as the thumbnail, or can rotate between the multiple.

> Why not include an "image" key so these NFTs don't break other places where NFTs are displayed?

i don't really understand what you mean here, could you elaborate more?

Reply:
fixed, it's specified as an array now